python/pyspark/testing/utils.py

```diff
@@ -336,7 +353,7 @@ def compare_vals(val1, val2):
                     and all(compare_vals(val1[k], val2[k]) for k in val1.keys())
                 )
             elif isinstance(val1, float) and isinstance(val2, float):
-                if abs(val1 - val2) > 1e-5:
+                if abs(val1 - val2) > (atol + rtol * abs(val2)):
```

Comment:
is it possible to directly use the `np.allclose` here?

maybe a few special values to consider:  float('nan'), float('inf'), float('-inf')